File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        left, right, ans = [0]*n, [0]*n, [0]*n

        left[0] = 1
        right[n-1] = 1

        for i in range(1,n):
            left[i] = left[i-1]*nums[i-1]

        for i in range(n-2,-1,-1):
            right[i] = right[i+1]*nums[i+1]

        for i in range(n):
            print(left[i]," ",right[i])
            ans[i] = left[i]*right[i]
        return ans

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int left = 0, right = nums.size()-1;
        while(right > left){
            int mid = left + ((right - left)/2);
            if(nums[right]>nums[mid]) right = mid;
            else left = mid+1;
        }
        return nums[left];
    }
};
